### PR TITLE
Increase max message size for rpc calls

### PR DESCRIPTION
**Pull request by komalali**
Fixes: https://github.com/pulumi/pulumi/issues/4155

Parent PR: https://github.com/pulumi/pulumi/pull/4455

### CODE REVIEW

File: sdk/nodejs/policy/server.ts [hunk 1]
```diff
@@ -63,6 +64,9 @@ let servingPolicyPack: string | undefined = undefined;
 // Regular expression for the policy pack name.
 const packNameRE = "^[a-zA-Z0-9-_.]{1,100}$";
 
+// Max message size for RPC calls
```
Comment (justinvp) [inline sdk/nodejs/policy/server.ts:67]: Use same comment as in https://github.com/pulumi/pulumi/pull/4455?

File: sdk/nodejs/policy/server.ts [hunk 2]
```diff
@@ -117,7 +121,9 @@ export function serve(
     servingPolicyPack = policyPackName;
 
     // Finally connect up the gRPC client/server and listen for incoming requests.
-    const server = new grpc.Server();
+    const server = new grpc.Server(<ChannelOptions>{
+        "grpc.max_receive_message_length": maxMessageSize,
+    });
```
Comment (justinvp) [inline sdk/nodejs/policy/server.ts:126]: Is the `<ChannelOptions>` cast necessary? Why not simply:

```typescript
    const server = new grpc.Server({
        "grpc.max_receive_message_length": maxMessageSize,
    });
```

?

File: sdk/python/lib/pulumi_policy/policy.py [hunk 3]
```diff
@@ -76,7 +78,10 @@ def __init__(self,
 
         servicer = _PolicyAnalyzerServicer(
             name, version, policies, enforcement_level if enforcement_level is not None else EnforcementLevel.ADVISORY)
-        server = grpc.server(futures.ThreadPoolExecutor(max_workers=4))
+        server = grpc.server(
+            futures.ThreadPoolExecutor(max_workers=4),
+            options=[('grpc.max_receive_message_length', _MAX_RPC_MESSAGE_SIZE)]
```
Comment (justinvp) [inline sdk/python/lib/pulumi_policy/policy.py:83]: Nit:

```suggestion
            options=[("grpc.max_receive_message_length", _MAX_RPC_MESSAGE_SIZE)]
```

File: sdk/python/lib/pulumi_policy/policy.py [hunk 4]
```diff
@@ -34,6 +34,8 @@
 
 _ONE_DAY_IN_SECONDS = 60 * 60 * 24
 
+_MAX_RPC_MESSAGE_SIZE = 1024 * 1024 * 400
```
Comment (justinvp) [inline sdk/python/lib/pulumi_policy/policy.py:37]: Comment?

File: sdk/nodejs/policy/server.ts (hunk 2 above)
Comment (komalali) [reply sdk/nodejs/policy/server.ts:126]: `grpc.Server()` expects the `ChannelOptions` type so I either need need to cast it in the invoke or declare it like:
```
const grpcChannelOptions: ChannelOptions = {
    "grpc.max_receive_message_length": maxMessageSize,
};
const server = new grpc.Server(grpcChannelOptions);
```

**Comment by komalali** [comment]
Thanks for the feedback @justinvp. I'll wait until https://github.com/pulumi/pulumi/pull/4455 is merged, make the changes from the PR and add the test and then ask you to re-review.

File: sdk/nodejs/policy/server.ts (hunk 2 above)
Comment (justinvp) [reply sdk/nodejs/policy/server.ts:126]: The TypeScript compiler actually complains if you don't specify the type?

Comment (justinvp) [reply sdk/nodejs/policy/server.ts:126]: I tried copying a minimal version of these types in a TypeScript [playground](https://www.typescriptlang.org/v2/en/play?#code/PTAEAsBdIBwZwFwgOYEtLgK4CMB0BjAewFthkAnGfMy-AWgDtCATAU2GwBtDtgB2AJzMAzPgBmrACysAbAEMAHMPnC5k7AEYADOK1apYgQFYjGhUYBMwzTK3AYc-AGs5yVnBpU6AKw9xy1PjgcgwMrJx0hDCQqIQMcLiQcADEADIWGnSpygBQrAAeMITkkKCoDJCs5GKOrKAAwsGh4QDy0bHxoADeOaB9oADkFFS4cHCcAPqQcuRukBMMcsSsE4QAblXkqGwDAPwIoHCQWwzIANy9-UO0uDBbxDMAnhOYcFUTrqwVewdHJ+eXPrXEZvIgMZhPF5vcgfNzffaHY7lAH9QbDfC4Ng1TCceZyTAYYroR4-RH-C6o4EYpysVgOTioDZTVDLCbEOCkhiYYjYKoUq7o3A0ulyBlMmLLQgEtkchFcnl8wFom7QtaofArMFiVDIUl-ZH8oGCh75CZg-CYcjkL7zP6sJayg7y3nkQ3KkbldCoUUTa1gsL4ebYRxOQhiMQyzncl1uqm4E2+1j+pNBkNhiPsqMK11KuOvFbcfA+uA4IIhMKTIqEThZmO541yU1vcFs9xwT4TThfZAYWuKykN01+1iMlbLMYdrunXty6P9-oAbRpj1+SNOAF1V-9QAAfUDOqq70CYcGsbVhZgXAC+ORyIAg0HgSDI6CweCIpHRnnoTDYHG4vCCCI4hSLIijKHIqjqNouj6JIhgmGYljWBotj2CGnweOiPh+AEwCqlUiQpKkGgaHweSFMUpT4JwchjKAADKVQbOQ3RKncjJyJUoBRDEcSIA0TQVm0fHxBcSpgn8mCBsUAAUvEdHACKNOWrTtPxACUbGoqiGCoAkCn8aAAC8PHqZ0uy7N0V5ujeN45JJpQmgAsm2nwMagABedSmdoFiSKAABUoB+QFwWSHo4mOYczGHqZYQAO6MbF5CyT0qIAESDomGqjq2E5uJ23YYBlBwuW5bged5AA0ORXhpFxAA), and the compiler lets me create the server instance without specifying the `ChannelOptions` type.

```typescript
// https://github.com/grpc/grpc-node/blob/79d3cfe4e6a836a3a4b10cf00e4f95518523b160/packages/grpc-js/src/channel-options.ts#L21-L36
export interface ChannelOptions {
    'grpc.ssl_target_name_override'?: string;
    'grpc.primary_user_agent'?: string;
    'grpc.secondary_user_agent'?: string;
    'grpc.default_authority'?: string;
    'grpc.keepalive_time_ms'?: number;
    'grpc.keepalive_timeout_ms'?: number;
    'grpc.service_config'?: string;
    'grpc.max_concurrent_streams'?: number;
    'grpc.initial_reconnect_backoff_ms'?: number;
    'grpc.max_reconnect_backoff_ms'?: number;
    'grpc.use_local_subchannel_pool'?: number;
    'grpc.max_send_message_length'?: number;
    'grpc.max_receive_message_length'?: number;
    [key: string]: string | number | undefined;
}

// https://github.com/grpc/grpc-node/blob/79d3cfe4e6a836a3a4b10cf00e4f95518523b160/packages/grpc-js/src/server.ts#L117
export class Server {
    private options: ChannelOptions;

    constructor(options?: ChannelOptions) {
        this.options = options ?? {};
    }
}

const maxMessageSize = 1024 * 1024 * 400;

const server = new Server({
    "grpc.max_receive_message_length": maxMessageSize,
});
```

Comment (komalali) [reply sdk/nodejs/policy/server.ts]: yeah you're right. I could've sworn it was giving me problems earlier 🤷‍♂️

**Comment by komalali** [comment]
@justinvp I added in the integration tests, wanna give it another quick glance please?

File: tests/integration/integration_test.go [hunk 5]
```diff
@@ -255,20 +255,21 @@ func runPolicyPackIntegrationTest(
...
-		},
-		{
-			WantErrors: []string{`enforcementLevel cannot be explicitly specified in properties.`},
-		},
-		{
-			WantErrors: []string{`"enforcementLevel" cannot be specified in required.`},
-		},
-	})
-}
+//
```
Comment (justinvp) [inline tests/integration/integration_test.go:258]: All the temporary commented-out tests need to be uncommented.

Comment (komalali) [reply tests/integration/integration_test.go:258]: hah thanks for catching that

Comment (justinvp) [reply tests/integration/integration_test.go:258]: Looks like this one test is still commented out.

Comment (komalali) [reply tests/integration/integration_test.go]: 🤦

**Pull request merged**